src/components/stop-card.ts

```diff
@@ -0,0 +1,150 @@
+import { Vue, Prop, Component } from 'vue-property-decorator';
+import { IStop } from '../api/stops';
+
+@Component
+export default class StopCard extends Vue {
+  // Stop to display
+  @Prop({
+    type: Object,
+    required: true,
+    default: () => ({})
+  })
+  readonly stop!: IStop;
+
+  // The inline-style applied to the card
+  style = {
+    bottom: '-100%',
+    transform: ''
+  };
+
+  // The min height for the card to be opened fully
+  openMin: number = 0;
+
+  // The max height for the card to be opened fully
+  openMax: number = 0;
+
+  // The min height for the card to be closed fully
+  closedMin: number = 0;
+
+  // The max height for the card to be closed fully
+  closedMax: number = 0;
+
+  // The initial Y position of the finger or touch
+  initialY!: number;
+
+  // The current Y position of the finger or touch
+  currentY!: number;
+
+  // The total Y offset of the finger or touch
+  yOffset: number = 0;
+
+  // Whether or not a dragging of the card is in progress
+  dragging: boolean = false;
+
+  /**
+   * Vue Lifecycle Hook
+   */
+  mounted() {
+    // The extra 8 px is the margin from the divider line
+    const heightOfTimetable = (this.$refs.stopTimetable as Vue).$el.clientHeight + 8;
+    const heightOfHeader = (this.$refs.stopDetails as Vue).$el.clientHeight;
+
+    // Set the default values based on the actual height of the cards
+    this.openMin = -heightOfTimetable / 2; // Should be 1/2 of the total height
+    this.openMax = -heightOfTimetable;
+    this.closedMin = heightOfHeader / 2; // Should be 1/2 of the total height
+    this.closedMax = heightOfHeader;
+
+    // Set the bottom, so that this will be the default state of the card
+    this.style = {
+      ...this.style,
+      bottom: `-${(this.$refs.stopTimetable as Vue).$el.clientHeight + 8}px`
+    };
+  }
+
+  /**
+   * Start the dragging process
+   * @param evt The touch or mouse drag event
```

Comment:
Does it have to be called `evt`? I like to use full descriptive words (usually) for vars

Reply:
Doesn't have to be, just my laziness in typing `event`. Will fix up :)